realm/src/main/java/io/realm/internal/Context.java

```diff
@@ -44,25 +44,50 @@
 
     public void executeDelayedDisposal() {
         synchronized (this) {
-            for (int i = 0; i < abandonedTables.size(); i++) {
-                long nativePointer = abandonedTables.get(i);
+            if(abandonedTables.size() > 0) {
+                for (long nativePointer : abandonedTables) {
```

Comment:
You should not use a `for...in` loop here as it will allocate a Iterator under the hood. It is safer to use:

```
int tableViewSize = abandonedTableViews.size()
 for (int i = 0; i < tableViewSize ; i++) {
  // ...
}
```